### PR TITLE
2022 09 na group2 bee system continued

### PR DESCRIPTION
**Pull request by LauWerUnity**
Bees pickup food and return it to base.
Some randomness and retargeting.
Works nicely for 2 factions.

### CODE REVIEW

File: Ported/combat-bees/Assets/Scripts/Systems/BeeSystem.cs [hunk 1]
```diff
@@ -93,7 +126,6 @@ public void Execute()
                     ECB.SetComponent(BeeEntities[i], new BeeProperties
                     {
                         BeeMode = BeeMode.FindResource,
-                        Aggressivity = BeeProperties[i].Aggressivity,
```
Comment (daniel-holz) [inline Ported/combat-bees/Assets/Scripts/Systems/BeeSystem.cs:96]: @LauWerUnity : is removing this here a mistake?

File: Ported/combat-bees/Assets/Scripts/Components/BeeProperties.cs [hunk 2]
```diff
@@ -13,5 +13,10 @@ public struct BeeProperties : IComponentData
         public float3 TargetPosition;
 
         public float Aggressivity;
+
+        // Exact world space spawn location (guaranteed in nest)
+        public float3 Origin;
```
Comment (daniel-holz) [inline Ported/combat-bees/Assets/Scripts/Components/BeeProperties.cs]: @LauWerUnity : I don't see you using that anywhere. Can we remove this?

**Review by daniel-holz — APPROVED** [review]

File: Ported/combat-bees/Assets/Scripts/Systems/BeeSystem.cs (hunk 1 above)
Comment (LauWerUnity) [reply Ported/combat-bees/Assets/Scripts/Systems/BeeSystem.cs:96]: Well I was thinking SetComponent only updates the fields you specify and ignores default values, that's probably wrong though, will re-add all the unchanged field copies and explicitly set them to past value, safer.

File: Ported/combat-bees/Assets/Scripts/Components/BeeProperties.cs (hunk 2 above)
Comment (LauWerUnity) [reply Ported/combat-bees/Assets/Scripts/Components/BeeProperties.cs]: Yes, forgot about that, applying corrections!

**Pull request merged**